src/stdlib_stats_distribution_PRNG.fypp

```diff
@@ -0,0 +1,212 @@
+    !! and real pseudorandom numbers
+    !!
+        ${t1}$, intent(in) :: n
+        ${t1}$ :: res
+        integer :: k
+
+        k = MAX_INT_BIT_SIZE - bit_size(n)
+        res = shiftr(xoshiro256ss( ), k)
+    end function dist_rand_${t1[0]}$${k1}$
+
+    #:endfor
+
+    function xoshiro256ss( ) result (res)
+    ! Generate random 64-bit integers
+    ! Written in 2018 by David Blackman and Sebastiano Vigna (vigna@acm.org)
+    ! http://prng.di.unimi.it/xoshiro256starstar.c
+    !
+    ! This is xoshiro256** 1.0, one of our all-purpose, rock-solid
+    ! generators. It has excellent (sub-ns) speed, a state (256 bits) that is
+    ! large enough for any parallel application, and it passes all tests we
+    ! are aware of.
+    !
+    ! The state must be seeded so that it is not everywhere zero. If you have
+    ! a 64-bit seed, we suggest to seed a splitmix64 generator and use its
+    ! output to fill st.
+    !
+    ! Fortran 90 version translated from C by Jim-215-Fisher
+    !
+        integer(int64) :: res, t
+
+        if(.not. seed_initialized) call random_distribution_seed_iint64(si,t)
+        res = rol64(st(2) * 5 , 7) * 9
+        t = shiftl(st(2), 17)
+        st(3) = ieor(st(3), st(1))
+        st(4) = ieor(st(4), st(2))
+        st(2) = ieor(st(2), st(3))
+        st(1) = ieor(st(1), st(4))
+        st(3) = ieor(st(3), t)
+        st(4) = rol64(st(4), 45)
+    end function xoshiro256ss
+
+    function rol64(x, k) result(res)
+        integer(int64), intent(in) :: x
+        integer, intent(in) :: k
+        integer(int64) :: t1, t2, res
+
+        t1 = shiftr(x, (64 - k))
+        t2 = shiftl(x, k)
+        res = ior(t1, t2)
+    end function rol64
+
+
+    subroutine jump
+    ! This is the jump function for the xoshiro256ss generator. It is equivalent
+    ! to 2^128 calls to xoshiro256ss(); it can be used to generate 2^128
+    ! non-overlapping subsequences for parallel computations.
+    ! Written in 2018 by David Blackman and Sebastiano Vigna (vigna@acm.org)
+    ! http://prng.di.unimi.it/xoshiro256starstar.c
+    !
+    ! Fortran 90 version translated from C by Jim-215-Fisher
+    integer(int64) :: jp(4) = [1733541517147835066_int64,                       &
+                               -3051731464161248980_int64,                      &
+                               -6244198995065845334_int64,                      &
+                               4155657270789760540_int64]
+    integer(int64) :: s1 = 0, s2 = 0, s3 = 0, s4 = 0, c = 1_int64
+    integer :: i, j, k
+
+    do i = 1, 4
+        do j = 1, 64
+            if(iand(jp(i), shiftl(c, j - 1)) /= 0) then
+                s1 = ieor(s1, st(1))
+                s2 = ieor(s2, st(2))
+                s3 = ieor(s3, st(3))
+                s4 = ieor(s4, st(4))
+            end if
+            k = xoshiro256ss( )
+        end do
+    end do
+    st(1) = s1
+    st(2) = s2
+    st(3) = s3
+    st(4) = s4
+    end subroutine jump
+
+    subroutine long_jump
```

Comment:
This function seems to be mainly for multi-process programs.

But what about mult-threading programming? As the module is now, I don't think `dist_rand` is suitable for multi-threading because, e.g., `st(:)` could be accessed simultaneously by multiple threads. Any thoughs on how to solve this?

For OpenMP, maybe the `threadprivate` directive could be useful to solve this problem?

Reply:
Multi-threading and multi-process have NOT been considered yet in all this probability distribution modules. The jump and long_jump functions included here are just to show the completeness provided by the original authors.

Reply:
Removed two procedures in this PR.

Reply:
> Multi-threading and multi-process have NOT been considered yet in all this probability distribution modules. 

If not done yet, please a note in the specs that multi-threading and multi-process have not been considered for these functions.